COURSE.md

```diff
@@ -1,38 +1,61 @@
 # Course outline
 
-### Low-level goals/topics
 This course is divided into separate modules, detailed below. Module A1-A4 are introductory, and contain what's needed to learn about the contents of the other modules. Therefore, you're advised to teach modules A1-A4 at the start of the course, but you can pick and mix the other modules.
 
-#### 0 - Course introduction
+## Learning objectives
+At the end of the course, the student is able to:
+- Describe the problems Rust aims to solve, and how (A1)
+- Identify the design goals of Rust (A1)
+- Decide whether or not Rust is a good fit for a project given requirements (A1)
+- Describe how Rust impacts professional contexts (A1)
+- Solve ownership errors using knowledge about the borrow checker model (A1, A3)
+- Use various tools to improve software quality and programming experience (A2)
+- Find their way around various Rust documentation sources (docs.rs, doc.rust-lang.org) and crate (dependency) indices (lib.rs) (A2)
+- Set up custom Rust application and library projects using Rusts build system and package manager (A2, B)
+- Implement small to medium-sized command-line applications using Rust and contribute to existing Rust projects (B)
+- Make informed choices on whether or not to use dependencies to implement functionality, and make a choice based on requirements (B)
+- Implement concurrent applications using threads and async/await (C)
```

Comment:
we deliberately chose *not* to cover async/await *yet* since we think our students will have enough problems with just understanding the concept of threading. that, combined with the fact that the async ecosystem is quickly changing made it difficult for us to lecture about it in a university setting. How do you aim to approach this?

Reply:
It seems to me that going into the internals of `async` would be unrealistic. However, there is a lot of production code already running on `tokio` and `async-std`. Glossing over the implications of `async` code and some examples of `tokio`-based applications should be enough to get started writing `async` applications. 

Maybe it's wise to state a couple of non-objectives as well, one of which would be learning about the async state machine, and implementing leaf futures.

Reply:
On further thought, I've decided to remove `async` from this module. @jonay2000 thanks for your feedback!